### PR TITLE
check the return value of BIO_new()

### PR DESCRIPTION
**Pull request by x2018**
Check the return value of BIO_new() in t_x509.c:471 & cmp_vfy.c:36 in time to prevent further memory errors.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

File: crypto/x509/t_x509.c [hunk 1]
```diff
@@ -470,6 +470,10 @@ int X509_STORE_CTX_print_verify_cb(int ok, X509_STORE_CTX *ctx)
         int cert_error = X509_STORE_CTX_get_error(ctx);
         BIO *bio = BIO_new(BIO_s_mem()); /* may be NULL */
 
+        if (bio == NULL) {
+            ERR_raise(ERR_LIB_X509, ERR_R_MALLOC_FAILURE);
```
Comment (t8m) [inline crypto/x509/t_x509.c:474]: An error is already raised by  BIO_new in case of malloc failure.

File: crypto/cmp/cmp_vfy.c [hunk 2]
```diff
@@ -34,7 +34,10 @@ static int verify_signature(const OSSL_CMP_CTX *cmp_ctx,
         return 0;
 
     bio = BIO_new(BIO_s_mem()); /* may be NULL */
-
+    if (bio == NULL) {
+        ERR_raise(ERR_LIB_CMP, ERR_R_MALLOC_FAILURE);
```
Comment (t8m) [inline crypto/cmp/cmp_vfy.c:38]: Same as below, this is not needed.

**Review by t8m — CHANGES_REQUESTED** [review]

File: crypto/cmp/cmp_vfy.c (hunk 2 above)
Comment (x2018) [reply crypto/cmp/cmp_vfy.c:38]: ok

**Review by t8m — APPROVED** [review]

**Review by slontis — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to 3.0 and master, thanks for the contribution.

**Pull request closed without merging**